pkg/utils/managedresources/managedresources.go

```diff
@@ -186,20 +189,43 @@ func deployManagedResource(ctx context.Context, secret *builder.Secret, managedR
 	return nil
 }
 
-// Delete deletes the managed resource and its secret with the given name in the given namespace.
-func Delete(ctx context.Context, client client.Client, namespace string, name string, secretNameWithPrefix bool) error {
+// Delete deletes the managed resource and its secrets with the given name in the given namespace.
+func Delete(ctx context.Context, c client.Client, namespace string, name string, secretNameWithPrefix bool) error {
 	secretName := SecretName(name, secretNameWithPrefix)
```

Comment:
This `secretName` is only required to deleted "old" secrets, isn't it? Once all secrets were migrated, we can probably drop this code as well. Can you add a TODO for that?